Sources/LiveKit/core/Engine.swift

```diff
@@ -193,11 +193,11 @@ class Engine: MulticastDelegate<EngineDelegate> {
     func addTrack(cid: String,
                   name: String,
                   kind: Livekit_TrackType,
-                  dimensions: Dimensions? = nil) -> Promise<Livekit_TrackInfo> {
+                  _ populator: (inout Livekit_AddTrackRequest) -> ()) -> Promise<Livekit_TrackInfo> {
```

Comment:
I think we can just pass a simple `dtx: Bool = true` to keep things simple ? 
like https://github.com/livekit/client-sdk-flutter/blob/721b04753a9e83e7815f694542b65db1f0eaea4d/lib/src/rtc_engine.dart#L149

Reply:
But interesting design is this to support many parameters in the future ?

Reply:
Hmm, yeah, was doing this to keep things clean so that we don't have to deal with random args polluting the method declaration (since audio args don't care about video args, and vice versa).

Reply:
Oh thats actually a good idea 👌